tests/ui/lint/decorate-def-path-str-ice.rs

```diff
@@ -0,0 +1,14 @@
+// Checks that compiling this file with
+// `-Dunused_must_use -Awarnings --cap-lints=warn --crate-type=lib` does not ICE when emitting
+// diagnostics.
```

Comment:
Seems a bit redundant given `compile-flags` below. It would probably be better to write some notes explaining why there is an ICE.

Reply:
I added some notes explaining why the ICE is triggered in the `tests/ui/lint/decorate-ice/decorate-def-path-str-ice.rs` test.